### DISCUSSION TITLE
Unable to open the video.

### DISCUSSION THREAD

**Discussion by EvgenieLebedev** [stella-cv/stella_vslam#524] [Q&A] [answered]
Hello! I took advantage of your guide, read the instructions and posted them in the "Best TL; DR User" section. Suggested to watch the build with the command: `./run_video_slam -v./orb_vocab.fbow -m ./aist_living_lab_1/video.mp4 -c ~/lib/stella_vslam/example/aist/equirectangular.yaml --frame-skip 3 --no-sleep --map-db-out map.msg**`

Immediately outputs digitization: The video cannot be opened.
terminate a call without an active exception

At first I thought that there was no file with the video, but I was able to view it and open it with third-party methods. 

Full logs:  
> Unable to open the video.
terminate a call without an active exception
Track scene (first last call last): 
#12 Object "", in 0xffffffffffffffffffffff, in
#11 Object "/root/lib/stella_vslam_examples/build/run_video_slam", in 0x56317f268004, in _start
#10 Source "./csu/../csu/libc-start.c", line 392, in __libc_start_main_impl [0x7f27cd829e3f]
#9 Source "./csu/../sysdeps/nptl/libc_start_call_main.h", line 58, in __libc_start_call_main [0x7f27cd829d8f]
#8 | Source "/root/lib/stella_vslam_examples/src/run_video_slam.cc", line 375, in the main
    | 374: will be returned back;
    | > 375: } 
    | Source "/usr/include/c++/11/bits/shared_ptr.h", line 122, in std::shared_ptr<stella_vslam::system>::shared_ptr()
    |   120:   */
    | 121: template<type name _Tp>
    | > 122: shared_ptr class : available __shared_ptr<_Tp>
    |   123:     {
    | 124: template<type name... _Args>
    | Source "/usr/include/c++/11/bits/shared_ptr_base.h", line 1154, in std::__shared_ptr<stella_vslam::system, (__gnu_cxx::_Lock_policy)2>::__shared_ptr()
| 1152: __shared_ptr(const __shared_ptr&) noexcept = default;
    | 1153: __shared_ptr& operator=(const __shared_ptr&) noexcept = default;
    | >1154: __shared_ptr() = by message;
    | 1155:
| 1156: template<type name _Yp, type name = _compatible<_Yp>>
      Source "/usr/include/c++/11/bits/shared_ptr_base.h", line 705, in std::__shared_count<(__gnu_cxx::_Lock_policy)2>::~__shared_count() [0x56317f2676f3]
        702: __total_quantity() without exception
        703:       {
        704: if (_M_pi != nullptr)
      > 705: _M_pi->_M_release();
        706:       }
        707:
        708: __shared_count(const __shared_count& __r) without exception
#7 Source "/usr/include/c++/11/bits/shared_ptr_base.h", line 168, in std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::_M_release() [0x56317f272d21]
        165: if (__gnu_cxx::__exchange_and_add_dispatch(&_M_use_count, -1) == 1)
166: {
167: _GLIBCXX_SYNCHRONIZATION_HAPPENS_AFTER(&_M_use_count);
      > 168: _M_dispose();
        169: // I can dispose() and destroy() for a long time to be the owner of memory
        170: // what to clean up, what objects to dispose of() are collected in a stream
171: //, which runs destroy().
#6 | Source "/root/lib/stella_vslam/src/stella_vslam/system.cc", line 106, in stella_vslam::system::~system()
| 105: system::~system() {
| > 106: global_optimization_thread_.sbros(nullptr);
    | 107: remove global_optimizer_;
    | 108: global_optimizer_ = nullptr;
    | Source "/usr/include/c++/11/bits/unique_ptr.h", line 456, in std::unique_ptr<std::thread, std::default_delete<std::thread> >::sbros(std::thread*)
    | 454: static_assert(__is_called by<deleter_type&, user>::meaning,
    | 455: "a means of removing unique attributes can be found free of charge for obtaining with the help of a pointer");
    | > 456: _M_t.reset(std::redefine(__p));
    |   457:       }
    | Source "/usr/include/c++/11/bits/unique_ptr.h", line 182, in std::__uniq_ptr_impl<std::thread, std::default_delete<std::thread> >::sbros(std::thread*)
    | 180: _M_ptr() = __p;
    | 181: if (__old_p)
| > 182: _M_deleter()(__old_p);
    |   183:       }
    | Source "/usr/include/c++/11/bits/unique_ptr.h", line 85, in std::default_delete<std::thread>::operator()(std::thread*) const
    | 83: static_assert(sizeof(_Tp)>0,
    | 84: "unable to delete pointer to incomplete type");
    | > 85: delete __ptr;
    |    86:       }
    |    87:     };
      Source "/usr/include/c++/11/bits/std_thread.h", line 152, in std::thread::~thread() [0x7f27cf6e5e86]
        149: stream()
150: {
151: if (attached())
> 152: std::register();
        153: }
154:
155: thread(const thread&) = delete;
#5 Object "/lib/x86_64-linux-gnu/libstdc++.so.6", in 0x7f27cdcae2b6, in std::terminate()
#4 Object "/lib/x86_64-linux-gnu/libstdc++.so.6", in 0x7f27cdcae24b, in std::rethrow_exception(std::__exception_ptr::exception_ptr)
#3 Object "/lib/x86_64-linux-gnu/libstdc++.so.6", in 0x7f27cdca2bbd, in __cxa_throw_bad_array_new_length
#2 Source "./stdlib/abort.c", line 79, in __GI_abort [0x7f27cd8287f2]
#1 Source "./signal/../sysdeps/posix/raise.c", line 26, in __GI_raise [0x7f27cd842475]
#0 | Source "./nptl/pthread_kill.c", line 89, in __GI___pthread_kill
    | Source "./nptl/pthread_kill.c", line 78, in __pthread_kill_internal
      Source "./nptl/pthread_kill.c", line 44, processed by __pthread_kill_ [0x7f27cd896a7c]
Previously (signal corrected by tkill() 182709 0)
Aborted (core reset)

**Comment by zoldaten** [comment, marked as answer]
i have the same issue with video files:
```
cd ~/lib/stella_vslam_examples/build/
./run_video_slam -v ./orb_vocab.fbow -m ./aist_living_lab_1/video.mp4 -c ~/lib/stella_vslam/example/aist/equirectangular.yaml --frame-skip 3 --no-sleep --map-db-out map.msg

[2023-08-22 18:18:07.734] [I] loading ORB vocabulary: ./orb_vocab.fbow
[2023-08-22 18:18:07.844] [I] load orb_params "default ORB feature extraction setting"
[2023-08-22 18:18:07.847] [I] startup SLAM system
[2023-08-22 18:18:07.848] [I] start mapping module
[2023-08-22 18:18:07.848] [I] start global optimization module
Unable to open the video.[
2023-08-22 18:18:07] [connect] Successful connection
Error: No active session
[2023-08-22 18:18:07] [connect] WebSocket Connection 127.0.0.1:3000 v-2 "WebSocket++/0.8.1" /socket.io/?EIO=4&transport=websocket&t=1692717487 101
[2023-08-22 18:18:07.853] [I] connected to server
```
but everything is OK with ROS streams from camera:
```
ros2 run stella_vslam_ros run_slam -v ~/lib/stella_vslam_examples/build/orb_vocab.fbow \
    -c ~/lib/stella_vslam/example/aist/equirectangular.yaml

[2023-08-22 18:22:48.316] [I] loading ORB vocabulary: /home/rock/lib/stella_vslam_examples/build/orb_vocab.fbow
[2023-08-22 18:22:48.353] [I] load orb_params "default ORB feature extraction setting"
[2023-08-22 18:22:48.354] [I] startup SLAM system
[2023-08-22 18:22:48.354] [I] start mapping module
[2023-08-22 18:22:48.354] [I] start global optimization module
[2023-08-22 18:22:48] [connect] Successful connection
[2023-08-22 18:22:48] [connect] WebSocket Connection 127.0.0.1:3000 v-2 "WebSocket++/0.8.1" /socket.io/?EIO=4&transport=websocket&t=1692717768 101
[2023-08-22 18:22:48.524] [I] connected to server
```

**Reply by EvgenieLebedev** [reply]
Thank you for paying attention to my problem. 
Over the past period I have been able to solve this problem in various ways rebuild: 
1) When I was building the program on a weak laptop, I did the following things to improve the stability of the work: 
- increased swap 
- instead of the "make -j" command, I used "make -j 2" or "make -j 3"
2) On a more powerful computer, I put the program under WSL2 with the help of a docker. 

I hope that these notes can be useful to other users :)
